components/supervisor/pkg/ports/tunneled-ports_test.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package ports
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/google/go-cmp/cmp"
+	"golang.org/x/sync/errgroup"
+)
+
+// TODO add reverse test
```

Comment:
```suggestion
// TODO(ak): add reverse test
```

(see https://www.notion.so/gitpod/How-we-develop-e8ce7e562478458a9223eb9b797415b2#86e669f8e7df4f1289c9bede0855b5f9)